spec/suites/layer/TooltipSpec.js

```diff
@@ -303,4 +303,14 @@ describe('Tooltip', function () {
 		layer.bindTooltip('Tooltip', {interactive: true});
 		layer.closeTooltip();
 	});
+
+	it("closes existent tooltip on new bindTooltip call", function () {
+		var layer = new L.Marker(center).addTo(map);
+		var eventSpy = sinon.spy(layer, "unbindTooltip");
+		layer.bindTooltip('Tooltip1', {permanent: true});
+		var tooltip1 = layer.getTooltip();
+		layer.bindTooltip('Tooltip2').openTooltip();
```

Comment:
```suggestion
		layer.bindTooltip('Tooltip2').openTooltip();
		layer.unbindTooltip.restore(); // unwrap the spy
```

Reply:
I will change it 👍 Only for understanding: is this not automatically removed / restored when the layer is removed / deleted by the test?

Reply:
Not sure, depends on implementation.
You could check by inspecting `L.Layer.prototype.unbindTooltip`

Reply:
`L.Layer.prototype.unbindTooltip` don't keeps a reference, so it would be in this case no problem without calling restore. But it is a cleaner code if it is called, so I added it.